### PR TITLE
Run ChainerX python tests in debug build

### LINKED ISSUE #6646: Run ChainerX python binding test in debug mode

### PR DESCRIPTION
**Pull request by niboshi**
Fixes #6646 

Depends on ~#6647~ and ~#6648~

### CODE REVIEW

**Comment by hvy** [comment]
#6648 has been merged.

**Comment by hvy** [comment]
Can you rebase this PR?

**Comment by niboshi** [comment]
Rebased

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
```
11:03:05 + pytest --showlocals --cov=chainerx --no-cov-on-fail --cov-fail-under=50 --cov-report html:/workspace/work/coverage-html/python /repo/tests/chainerx_tests
11:03:06 ============================= test session starts ==============================
11:03:06 platform linux -- Python 3.6.8, pytest-4.1.1, py-1.8.0, pluggy-0.9.0
11:03:06 rootdir: /repo, inifile: setup.cfg
11:03:06 plugins: cov-2.6.1
11:03:46 collected 71094 items
11:03:46 
11:03:46 repo/tests/chainerx_tests/test_array_utils.py .......................... [  0%]
11:03:46 ........................................................................ [  0%]

:

11:11:50 s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.sssssssssssss.s.s.s.s.s.s.s.s.s.s.s.s. [ 94%]
11:11:52 s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.s.ss [ 94%]
11:11:54 sssssssssss./repo/scripts/ci/steps.sh: line 190:  2052 Aborted                 (core dumped) COVERAGE_FILE="$WORK_DIR"/coverage-data pytest --showlocals --cov=chainerx --no-cov-on-fail --cov-fail-under=50 --cov-report html:"$WORK_DIR"/coverage-html/python "$REPO_DIR"/tests/chainerx_tests
```

File: chainerx_cc/chainerx/cuda/cuda_device/batch_norm.cc [hunk 1]
```diff
@@ -312,10 +309,9 @@ Array CudaDevice::FixedBatchNorm(
         CHAINERX_ASSERT(&x.device() == &mean.device());
         CHAINERX_ASSERT(&x.device() == &var.device());
 
-        CHAINERX_ASSERT(x.dtype() == gamma.dtype());
-        CHAINERX_ASSERT(x.dtype() == beta.dtype());
-        CHAINERX_ASSERT(x.dtype() == mean.dtype());
-        CHAINERX_ASSERT(x.dtype() == var.dtype());
+        CHAINERX_ASSERT(gamma.dtype() == beta.dtype());
+        CHAINERX_ASSERT(gamma.dtype() == mean.dtype());
+        CHAINERX_ASSERT(gamma.dtype() == var.dtype());
```
Comment (hvy) [inline chainerx_cc/chainerx/cuda/cuda_device/batch_norm.cc]: I think this can be relaxed to check the kind as in BatchNorm https://github.com/chainer/chainer/pull/6649/files#diff-aad3fa437c706c597c3720da0a68f7a0R139 It is cast anyway and it's technically supported?

**Comment by hvy** [comment]
I left a smaller comment, could you take a look?

**Comment by niboshi** [comment]
Fixed

**Comment by hvy** [comment]
Jenkins, test this please.

**Comment by hvy** [comment]
CI failures seem unrelated. LGTM!
